app/models/loan_repayment.rb

```diff
@@ -66,7 +66,7 @@ class LoanRepayment < Ekylibre::Record::Base
   }
 
   before_validation do
-    self.amount = base_amount + insurance_amount + interest_amount
+    self.amount = base_amount + insurance_amount + interest_amount rescue 0
```

Comment:
Can you avoid use of `rescue`. Instead of that, can you add a default assignment on each amount (`xxx_amount ||= 0`)